src/kOS/Module/kOSVesselModule.cs

```diff
@@ -223,16 +226,55 @@ private void HookEvents()
         /// </summary>
         private void UnHookEvents()
         {
-            if (RemoteTechHook.IsAvailable(parentVessel.id))
+            if (hasRemoteTech)
             {
-                RemoteTechHook.Instance.RemoveSanctionedPilot(parentVessel.id, UpdateAutopilot);
+                UnHookRemoteTechPilot();
             }
             else
             {
-                parentVessel.OnPreAutopilotUpdate -= UpdateAutopilot;
+                UnHookStockPilot();
             }
         }
 
+        private void CheckSwapEvents()
```

Comment:
```
/// <summary>
/// A race condition exists when KSP can load the kOS module onto the vessel
/// before it loaded the RemoteTech module.  That makes it so that kOS may not
/// see the existence of RT yet when kOS is first initialized.<br/>
/// <br/>
/// This fixes that case by continually re-querying for RT post-loading, and
/// re-initializing kOS's RT-related behaviors if it seems that the RT module
/// now exists when it didn't before (or visa versa).
/// </summary>
```
I changed a couple of tenses above because it read a little funny to me identifying the race condition in past tense, when the condition still exists we just catch it better.

I was fully planning on doing comments tonight before you merge.  But if you think this summary is enough, then I'm good with it.  That gives me more time to focus on merging PR's instead.